src/point.rs

```diff
@@ -35,15 +35,21 @@ impl Point {
 
     /// Linearly interpolate between two points.
     #[inline]
-    pub fn lerp(&self, other: Point, t: f64) -> Point {
+    pub fn lerp(self, other: Point, t: f64) -> Point {
         self.to_vec2().lerp(other.to_vec2(), t).to_point()
     }
 
     /// Determine the midpoint of two points.
     #[inline]
-    pub fn midpoint(&self, other: Point) -> Point {
+    pub fn midpoint(self, other: Point) -> Point {
         Point::new(0.5 * (self.x + other.x), 0.5 * (self.y + other.y))
     }
+
+    /// Euclidean distance.
+    #[inline]
+    pub fn distance(self, other: Point) -> f64 {
```

Comment:
This is probably better written as simply `(self - other).hypot()`. Note that the `powf(2.)` is not the most efficient way to square, `powi(2)` is probably better (although I wouldn't be surprised if llvm were smart enough to optimize this anyway).